newscoop/src/Newscoop/NewscoopBundle/Resources/public/js/playlists/controllers/playlists.js

```diff
@@ -115,6 +120,83 @@ angular.module('playlistsApp').controller('PlaylistsController', [
         }
     });
 
+    // stops, starts counter
+    $scope.isCounting = false;
+
+    /**
+     * This function count seconds after which revert popup will be closed
+     * and removes last article from the playlist if the limit is reached,
+     * inserts new article
+     */
+    var countDown = function(){
+       if($scope.isCounting) {
+            return;
+       }
+
+       $scope.isCounting = true;
+       (function countEvery() {
+            if ($scope.isCounting) {
+                $scope.countDown--;
+                $timeout(countEvery, 1000);
+                if ($scope.countDown === 0) {
+                    removeLastArticle();
+                }
+            }
+        }());
+    }
+
+    /**
+     * It removes last article from the playlist if the limit is reached,
+     * inserts new article. It is called from FeaturedController.
+     */
+    $scope.removeLastInsertNew = function () {
+        removeLastArticle();
+    }
+
+    /**
+     * It removes last article from the playlist if the limit is reached,
+     * inserts new article
+     */
+    var removeLastArticle =  function () {
+        // remove one before last article from the featured articles list
+        // when we drag-drop new article, it will be
+        // automatically added to the list as a last element, thats why we need to remove
+        // one before last article
+        var articleToRemove = $scope.featuredArticles[$scope.featuredArticles.length - 2];
+        articleToRemove._method = "unlink";
+        _.remove(
```

Comment:
no, I remove it here only from the playlist which user loads (right column) so it's removed from the array only, then a proper info is added to the array were I store all of the actions and on list save its all performed.